stdlib/public/core/Shims.swift

```diff
@@ -41,3 +41,23 @@ internal let _fastEnumerationStorageMutationsPtr =
 internal func _mallocSize(ofAllocation ptr: UnsafeRawPointer) -> Int? {
   return _swift_stdlib_has_malloc_size() ? _swift_stdlib_malloc_size(ptr) : nil
 }
+
+/*
+ Invariant:
+ malloc_size(malloc(malloc_good_size(size))) >= malloc_good_size(size)
+ 
+ Usually:
+ malloc_size(malloc(malloc_good_size(size))) == malloc_good_size(size)
+ */
+@inlinable @_effects(readnone)
+internal func _mallocGoodSize(for size: Int) -> Int {
+  if (size <= 256) {
+    return (size &+ 15) & ~15;
+  }
```

Comment:
Should we avoid encoding this allocator assumption?  Can we measure if this is optimization is worth it?

Reply:
I'll try the experiment both ways, have to get it working first though

Reply:
oh also now that I'm not doing this for array I can stop marking this inlinable, since it's only called from non-inlinable code